test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -747,6 +747,29 @@ describe('Loader hooks', { concurrency: true }, () => {
     assert.strictEqual(signal, null);
   });
 
+  it('should support source maps in commonjs translator', async () => {
+    const readFile = async () => {};
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--enable-source-maps',
+      '--experimental-loader',
+      `data:text/javascript,import{readFile}from"node:fs/promises";export ${
+        async function load(u, c, n) {
+          const r = await n(u, c);
+          if (c.format === 'commonjs') {
+            r.source = await readFile(new URL(u));
+          }
+          return r;
```

Comment:
Can you please spell out these single-character variable names?

Reply:
I followed the precedent from the test above. I think I should do that holistically in different PR.

Reply:
Let's not add more tests that are hard to decipher. Just because some other test isn't as legible as it can be doesn't mean we should add another one.

Reply:
Same response: https://github.com/nodejs/node/pull/51033#discussion_r1414826453